path/windows/normalize.ts

```diff
@@ -23,7 +24,38 @@ import { isPathSeparator, isWindowsDeviceRoot } from "./_util.ts";
  * @param path The path to normalize
  * @returns The normalized path
  */
-export function normalize(path: string): string {
+export function normalize(path: string): string;
+/**
+ * Normalize the `path`, resolving `'..'` and `'.'` segments.
+ * Note that resolving these segments does not necessarily mean that all will be eliminated.
+ * A `'..'` at the top-level will be preserved, and an empty path is canonically `'.'`.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @example Usage
+ * ```ts
+ * import { normalize } from "@std/path/windows/normalize";
+ * import { assertEquals } from "@std/assert";
+ *
+ * const normalized = normalize("C:\\foo\\..\\bar");
+ * assertEquals(normalized, "C:\\bar");
+ * ```
+ *
+ * @example Working with URLs
+ * ```ts
+ * import { normalize } from "@std/path/windows/normalize";
+ * import { assertEquals } from "@std/assert";
+ *
+ * const normalized = normalize(new URL("file:///C:/foo/../bar"));
+ * assertEquals(normalized, "C:\\bar");
+ * ```
```

Comment:
```suggestion
 * @example Usage
 * ```ts
 * import { normalize } from "@std/path/windows/normalize";
 * import { assertEquals } from "@std/assert";
 *
 * const normalized = normalize(new URL("file:///C:/foo/../bar"));
 * assertEquals(normalized, "C:\\bar");
 * ```
```